lib/Controller/SocialApiController.php

```diff
@@ -0,0 +1,294 @@
+	 * @returns {array} an array of supported social networks
+	 */
+	public function getSupportedNetworks(string $type) : ?array {
+
+		$supported = array();
+		$supported['avatar'] = array();
+
+		foreach(self::SOCIAL_CONNECTORS as $network => $social) {
+			array_push($supported['avatar'], $network);
+		}
+
+		if (strcmp($type, 'all') === 0) {
+			// return array of arrays
+			return $supported;
+		}
+		if (array_key_exists($type, $supported)) {
+			return $supported[$type];
+		}
+		// unknown type
+		return array();
+	}
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * generate download url for a social entry (based on type of data requested)
+	 *
+	 * @param {array} socialentry the network and id from the social profile
+	 * @returns {String} the url to the requested information or null in case of errors
+	 */
+	protected function getSocialConnector(array $socialentry) : ?string {
+
+		$connector = null;
+
+		// check supported networks in order
+		foreach(self::SOCIAL_CONNECTORS as $network => $social) {
+
+			// search for this network in user's profile
+			foreach ($socialentry as $networkentry => $profileId) {
+				if ($network === strtolower($networkentry)) {
+					// cleanups
+					if (in_array('basename', $social['cleanups'])) {
+						$profileId = basename($profileId);
+					}
+					if (in_array('filter', $social['cleanups'])) {
+						if (preg_match($social['filter']['regex'], $profileId, $matches)) {
+							$profileId = $matches[$social['filter']['group']];
+						}
+					}
+					// checks
+					if (in_array('number', $social['checks'])) {
+						if (!ctype_digit($profileId)) {
+							break;
+						}
+					}
+					$connector = str_replace("{socialId}", $profileId, $social['recipe']);
+					break;
+				}
+			}
+			if ($connector) {
+				break;
+			}
+		}
+		return ($connector);
+	}
+
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * Retrieves social profile data for a contact
+	 *
+	 * @param {String} addressbookId the addressbook identifier
+	 * @param {String} contactId the contact identifier
+	 * @param {String} type the kind of information to retrieve -- provision
+	 *
+	 * @returns {JSONResponse} an empty JSONResponse with respective http status code
+	 */
+	public function fetch(string $addressbookId, string $contactId, string $type) : JSONResponse {
+
+		$url = null;
+
+		try {
+			// get corresponding addressbook
+			$addressBooks = $this->manager->getUserAddressBooks();
+			$addressBook = null;
+			foreach($addressBooks as $ab) {
+				if ($ab->getUri() === $addressbookId) {
+					$addressBook = $ab;
+				}
+			}
+			if (is_null($addressBook)) {
+				return new JSONResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
+			}
+
+			// search contact in that addressbook
+			$contact = $addressBook->search($contactId, ['UID'], [])[0];
+			if (is_null($contact)) {
+				return new JSONResponse([], Http::STATUS_INTERNAL_SERVER_ERROR); 
+			}
+
+			// get social data
+			$socialprofile = $contact['X-SOCIALPROFILE'];
```

Comment:
This can/should be an array!
You don't know which property to sync from.
See our old discussion with https://github.com/nextcloud/server/pull/20529

Reply:
https://github.com/nextcloud/contacts/pull/1580#discussion_r413128343

Reply:
Yes, this is an array. I renamed the variable to make it clear.